### PR TITLE
Best Practices

### PR DESCRIPTION
**Pull request by omkar-ti**
@kushal-ti Please review

- [x] - Removed `import *`
- [x] - Modified imports where only a single function was required
- [x] - Modified a few class names and function names to follow convention
- [x] - Put the secret key in .env file, .env file is present in gitignore
- [x] - Added type hints and annotations wherever possible
- [x] - Replaced numerical HTTP status codes with provided enums
- [x] - Removed commented code

### CODE REVIEW

**Comment by omkar-ti** [comment]
- Removed `import *`
- Modified imports where only a single function was required
- Put the secret key in .env file, .env file is present in gitignore
- Added type hints and annotations wherever possible
- Replaced numerical HTTP status codes with provided enums

**Comment by Smith-TI** [comment]
As requested by @kushal-ti I'm taking up this PR for a preliminary review before his final review

**Comment by Smith-TI** [comment]
Nit: Use [Github Tasklist](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists) for a better UI for PRs with multiple tasks

File: manage.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-import sys
+import os.environ.setdefault
+import sys.argv
 from django.core.management.commands.runserver import Command as runserver
+from cjapp.settings import DEFAULT_PORT
 
 if __name__ == "__main__":
     os.environ.setdefault("DJANGO_SETTINGS_MODULE", "cjapp.settings")
-    runserver.default_port = "8080"
+    runserver.DEFAULT_PORT = DEFAULT_PORT
```
Comment (Smith-TI) [inline manage.py:9]: `DEFAULT_PORT` doesn't exist on runserver

```suggestion
    runserver.default_port = DEFAULT_PORT
```

File: restapi/views.py [hunk 2]
```diff
@@ -190,22 +190,22 @@ def logProcessor(request):
...
-    logs = multiThreadedReader(urls=data['logFiles'], num_threads=data['parallelFileProcessingCount'])
+    logs = multi_threaded_reader(urls=data['logFiles'], num_threads=data['parallelFileProcessingCount'])
     sorted_logs = sort_by_time_stamp(logs)
     cleaned = transform(sorted_logs)
     data = aggregate(cleaned)
     response = response_format(data)
     return Response({"response":response}, status=status.HTTP_200_OK)
 
-def sort_by_time_stamp(logs):
+def sort_by_time_stamp(logs) -> list:
```
Comment (Smith-TI) [inline restapi/views.py:200]: Can be moved to another file (since this is not a view, but a utility function)

File: restapi/views.py [hunk 3]
```diff
@@ -259,7 +259,7 @@ def reader(url, timeout):
         return conn.read()
 
 
-def multiThreadedReader(urls, num_threads):
+def multi_threaded_reader(urls, num_threads) -> list:
```
Comment (Smith-TI) [inline restapi/views.py:262]: Instead of fetching the URLs one by one, fetch them concurrently

File: restapi/views.py [hunk 4]
```diff
@@ -180,7 +180,7 @@ def get_queryset(self):
 @api_view(['post'])
 @authentication_classes([])
 @permission_classes([])
-def logProcessor(request):
+def logProcessor(request) -> Response:
```
Comment (Smith-TI) [inline restapi/views.py:183]: 1. Rename the function
2. Add Logging to determine the time taken to process logs

File: restapi/serializers.py [hunk 5]
```diff
@@ -41,7 +41,7 @@ class Meta(object):
 class ExpensesSerializer(ModelSerializer):
     users = UserExpenseSerializer(many=True, required=True)
 
-    def create(self, validated_data):
+    def create(self, validated_data) -> Expenses:
```
Comment (Smith-TI) [inline restapi/serializers.py:44]: Add doctrings to describe what the function does

**Review by Smith-TI — CHANGES_REQUESTED** [review]
Preliminary Review 1

**Comment by omkar-ti** [comment]
@Smith-TI I have tried doing the changes as suggested.

**Pull request closed without merging**